flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,273 @@
+    private static final List<Row> trainArray =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of(1, Vectors.dense(2.0, 3.0)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(2, Vectors.dense(200.1, 300.1)),
+                            Row.of(2, Vectors.dense(200.2, 300.2)),
+                            Row.of(2, Vectors.dense(200.3, 300.3)),
+                            Row.of(2, Vectors.dense(200.4, 300.4)),
+                            Row.of(2, Vectors.dense(200.4, 300.4)),
+                            Row.of(2, Vectors.dense(200.6, 300.6)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.3, 3.2)),
+                            Row.of(1, Vectors.dense(2.3, 3.2)),
+                            Row.of(3, Vectors.dense(2.8, 3.2)),
+                            Row.of(4, Vectors.dense(300., 3.2)),
+                            Row.of(1, Vectors.dense(2.2, 3.2)),
+                            Row.of(5, Vectors.dense(2.4, 3.2)),
+                            Row.of(5, Vectors.dense(2.5, 3.2)),
+                            Row.of(5, Vectors.dense(2.5, 3.2)),
+                            Row.of(1, Vectors.dense(2.1, 3.1))));
+
+    private static final List<Row> testArray =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of(5, Vectors.dense(4.0, 4.1)), Row.of(2, Vectors.dense(300, 42))));
+    private Table testData;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+
+        Schema schema =
+                Schema.newBuilder()
+                        .column("f0", DataTypes.INT())
+                        .column("f1", DataTypes.of(DenseVector.class))
+                        .build();
+
+        DataStream<Row> dataStream = env.fromCollection(trainArray);
+        trainData = tEnv.fromDataStream(dataStream, schema).as(LABEL_COL + "," + VEC_COL);
+
+        DataStream<Row> predDataStream = env.fromCollection(testArray);
+        testData = tEnv.fromDataStream(predDataStream, schema).as(LABEL_COL + "," + VEC_COL);
+    }
+
+    // Executes the graph and returns a list which has true label and predict label.
+    private static List<Tuple2<String, String>> executeAndCollect(Table output) throws Exception {
```

Comment:
Should the output type be `List<Tuple2<Integer, Integer>>`? 

Could you help check that other usages of `String` has been corrected in `KnnTest`?

Reply:
OK